Add splunk-otel-go example (#106)
signalfx/tracing-examples: opentelemetry-tracing/opentelemetry-go/README.md

> # The Splunk distribution of OpenTelemetry Go instrumentation
> 
> This repository contains some example applications that utilize auto-instrumentation for
> a supported Go HTTP libraries.
> 
> For more detailed information about the functionality of the provided tracer and instrumentation configuration,
> please see the [Splunk distribution of OpenTelemetry Go instrumentation project](https://github.com/signalfx/splunk-otel-go).
> 
> ## Libraries and Framework Examples

Reviewer: ```suggestion
# Splunk distribution of OpenTelemetry Go

This repository contains sample applications that use autoinstrumentation for
supported Go HTTP libraries.

For more information on the tracer and instrumentation configuration, see the
[Splunk distribution of OpenTelemetry Go instrumentation project](https://github.com/signalfx/splunk-otel-go).

## Libraries and framework examples
```

Author: ```suggestion
# Splunk distribution of OpenTelemetry Go

This repository contains sample applications that use instrumentation for
supported Go HTTP libraries.

For more information on the tracer and instrumentation configuration, see the

[Splunk distribution of OpenTelemetry Go instrumentation project](https://github.com/signalfx/splunk-otel-go).

## Libraries and framework examples
```
